services/workflows-service/src/data-migration/scripts/migrate.ts

```diff
@@ -0,0 +1,152 @@
+import { DataMigrationVersion, PrismaClient } from '@prisma/client';
+import * as path from 'path';
+import * as fs from 'fs';
+import { INestApplicationContext } from '@nestjs/common';
+import { env } from '@/env';
+import { DataMigrationRepository } from '@/data-migration/data-migration.repository';
+import { isErrorWithMessage } from '@ballerine/common';
+import console from 'console';
+import { DATA_MIGRATION_FOLDER_RELATIVE_PATH } from './consts';
+
+interface IMigrationProcess {
+  version: string; // also the timestamp
+  migrate: (client: PrismaClient, app: INestApplicationContext) => Promise<void>;
+  fileName: string;
+}
+
+const fetchMigrationFiles = async (migrationFolderPath: string) => {
+  if (!fs.existsSync(migrationFolderPath)) {
+    console.log(`Missing Migration Folder ${migrationFolderPath}`);
+    return [];
+  }
+
+  const fullFilePath = (await fs.promises.readdir(migrationFolderPath)).map(fileName =>
+    path.join(migrationFolderPath, fileName),
+  );
+
+  return fullFilePath;
+};
+
+const fetchEnvMigrationFiles = async () => {
+  let migrationFiles: Array<string> = [];
+  const commonMigrations = path.join(__dirname, DATA_MIGRATION_FOLDER_RELATIVE_PATH, 'common');
+  migrationFiles = migrationFiles.concat(await fetchMigrationFiles(commonMigrations));
+
+  const environmentName = env.ENVIRONMENT_NAME || '';
+  if (['local', 'dev', 'development', 'demo'].includes(environmentName)) {
+    migrationFiles = migrationFiles.concat(
+      await fetchMigrationFiles(path.join(__dirname, DATA_MIGRATION_FOLDER_RELATIVE_PATH, 'dev')),
+    );
+  }
+
+  if (['sandbox', 'sb'].includes(environmentName)) {
+    migrationFiles = migrationFiles.concat(
+      await fetchMigrationFiles(path.join(__dirname, DATA_MIGRATION_FOLDER_RELATIVE_PATH, 'sb')),
+    );
+  }
+
+  if (['prod', 'production'].includes(environmentName)) {
+    migrationFiles = migrationFiles.concat(
+      await fetchMigrationFiles(path.join(__dirname, DATA_MIGRATION_FOLDER_RELATIVE_PATH, 'prod')),
+    );
+  }
+
+  return migrationFiles;
+};
+
+export const migrate = async () => {
+  const client = new PrismaClient();
+  const appContext = await NestFactory.createApplicationContext(AppModule);
+
+  const dataMigrationRepository = appContext.get(DataMigrationRepository);
+  const migrationFilesPath = await fetchEnvMigrationFiles();
+
+  const latestSuccessfulMigration = await dataMigrationRepository.getLatestTimestamp();
+  const migrationProcessesToRun = await calculateMigrationProcessToRun(
+    migrationFilesPath,
+    latestSuccessfulMigration as DataMigrationVersion,
+  );
+
+  for (const migrationProcess of migrationProcessesToRun) {
+    const migrationVersion = migrationProcess.version;
+    console.log(
+      `Running Data Migration: ${migrationProcess.fileName.split('/').pop()!.replace('.js', '')}`,
+    );
+    try {
+      await migrationProcess.migrate(client, appContext);
+
+      await dataMigrationRepository.create({
```

Comment:
Its safer to create this record before the migration runs, as "processing" state and then update it to completed or failed - if for any reason the migration hangs or crash the pod we can see it stuck on the db as "processing"